packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -25,8 +32,16 @@ import BlockInspectorButton from './block-inspector-button';
 import PluginBlockSettingsMenuGroup from '../block-settings-menu/plugin-block-settings-menu-group';
 
 function VisualEditor() {
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const isFocusMode = useSelect( ( select ) =>
+		select( 'core/edit-post' ).isFeatureActive( 'focusMode' )
+	);
+	const className = classnames( 'edit-post-visual-editor editor-styles-wrapper', {
+		'is-focus-mode': isLargeViewport && isFocusMode,
+	} );
```

Comment:
Moving the class here means for me that the focus mode is not a "core" feature of the generic "block-editor" package and that is something added specifically by "edit-post". Is this what we really want? Do we want third-party block editors to also support focus mode?

Depending on the answer we might have to change several things:

 - If yes, move focus mode styles to the edit-post package and potentially deprecate/remove the focusMode setting in the block-editor settings.
 - If no, keep the styles in the block-editor package but move this class to "BockList" or something like that.

Reply:
I think that it shouldn't be part of `block-editor` as it's a feature controlled by `edit-post` and should be isolated within that component. I do agree that it would be good to move the styles there as well.

> Do we want third-party block editors to also support focus mode?

They should explicitly add support, yes. They already need to do that (by adding a control for it).

Reply:
> They already need to do that (by adding a control for it).

That's not true, they can decide to enable it by default in the settings and this PR is breaking that.

I don't think we should use the fact that the toggle is in edit-post as an argument to decide whether this feature should be in block-editor or not. 

I'm personally thinking it does make sense there otherwise, people will find it hard to duplicate these styles and recreate them while it can be offered by default with a config.

Reply:
Ah, I see what you mean. They could control the feature though the editor settings. I'm fine with adding the class to the root block list instead of here.